include/kodlab_mjbots_sdk/mjbots_simulation_interface.h

```diff
@@ -0,0 +1,246 @@
+// BSD 3-Clause License
+// Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+// Authors:
+// Shane Rozen-Levy <srozen01@seas.upenn.edu>
+#ifndef _mjbots_simulation_interface
+#define _mjbots_simulation_interface 
+
+#pragma once
+
+#include <vector>
+#include <map>
+#include <future>
+#include <memory>
+#include <optional>
+#include "kodlab_mjbots_sdk/moteus_protocol.h"
+#include "kodlab_mjbots_sdk/joint_moteus.h"
+#include "kodlab_mjbots_sdk/pi3hat_moteus_interface.h"
+#include "kodlab_mjbots_sdk/soft_start.h"
+#include "kodlab_mjbots_sdk/imu_data.h"
+#include "kodlab_mjbots_sdk/robot_interface.h"
+
+#include <iostream>
+#include <mujoco/mjmodel.h>
+#include <mujoco/mujoco.h>
+
+#include "cstdio"
+#include "cstdlib"
+#include "cstring"
+#include <GLFW/glfw3.h>
+// Libraries for sleep
+#include <chrono>
+#include <thread>
+
+
+namespace kodlab::mjbots {
+
+
+
+/*!
+ * @brief Object allowing interaction with the Mjbots Moteus motor controller
+ *        hardware
+ */
+class MjbotsSimulationInterface : public RobotInterface {
+ public:
+
+  /*!
+   * @brief constructs an mjbots_robot_interface to communicate with a collection of moeteusses
+   * @param joint_list a vector of shared pointers to joints defining the motors in the robot
+   * @param realtime_params the realtime parameters defining cpu and realtime priority
+   * @param imu_mounting_deg Orientation of the imu on the pi3hat. Assumes gravity points in the +z direction
+   * @param imu_rate_hz Frequency of the imu updates from the pi3hat
+   * @param imu_data_ptr Shared pointer to imu_data to use or nullptr if it should make its own
+   * @param imu_world_offset_deg [Optional] IMU orientation offset. Useful for re-orienting gravity, etc.
+   * @param dry_run if true, sends zero-torques to Moteus controllers
+   * @param print_torques if true, prints torque commands
+   * @param send_pd_commands if true, packets to the moteus include pd gains and setpoints
+   */
+  MjbotsSimulationInterface(std::vector<std::shared_ptr<JointMoteus>> joint_list,
+                       const RealtimeParams &realtime_params,
+                       ::mjbots::pi3hat::Euler imu_mounting_deg = ::mjbots::pi3hat::Euler(),
+                       int imu_rate_hz = 1000,
+                       std::shared_ptr<::kodlab::IMUData<float>> imu_data_ptr = nullptr,
+                       std::optional<::mjbots::pi3hat::Euler > imu_world_offset_deg = std::nullopt,
```

Comment:
LATER, we should fix this mjbots::pi3hat imu include type 
IDEALLY no mjbots code is necessary.
This probably has ramifications I can't see now, and therefore should be solved later